pep-0458.txt

```diff
@@ -875,17 +859,22 @@ was susceptible to these attacks and how TUF could protect users against them
 
 __ https://mail.python.org/pipermail/distutils-sig/2013-September/022755.html
 
-Note that compromising *targets* or any delegated role (except for project
-targets metadata) does not immediately allow an attacker to serve malicious
+Note that compromising *targets* or *bins*
+does not immediately allow an attacker to serve malicious
 updates.  The attacker must also compromise the *timestamp* and *snapshot*
 roles, which are both online and therefore more likely to be compromised.
 This means that, in order to launch any attack, one must not only be able to
 act as a man-in-the-middle, but also compromise the *timestamp* key (or
 compromise the *root* keys and sign a new *timestamp* key).  To launch any
 attack other than a freeze attack, one must also compromise the *snapshot* key.
+In practice, this PEP recommends storing the *snapshot*, *timestamp*, and 
+*bin-n* keys together, or even using the same key for all of these roles.
+Because of this, the attacker only needs to compromise this single server to
+perform any of the attacks listed above. However, the offline *root* key will
```

Comment:
We should also not undersell. Let's stress that the compromise must be on the PyPI signing infrastructure and not on a CDN or mirror. Something along the lines of....

```suggestion
perform any of the attacks listed above. Note that clients are still protected
against compromises of non-signing infrastructure such as CDNs or mirrors.
Moreover, the offline *root* key will
```